[Crash] EOF while retrieving data azurerm_virtual_network

Hi,

Adding the following:

```
data "azurerm_virtual_network" "elastic" {
  resource_group_name = "Prod-Net-Arm"
  name = "prod-arm-vnet-general-us1"
}
```

To my Terraformed project resulted in the following crash

```
Error: Error refreshing state: 3 error(s) occurred:

* data.azurerm_virtual_network.elastic: 1 error(s) occurred:

* data.azurerm_virtual_network.elastic: data.azurerm_virtual_network.elastic: unexpected EOF

panic: runtime error: invalid memory address or nil pointer dereference
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13c1c6b]
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: 
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: goroutine 145 [running]:
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.dataSourceArmVnetRead(0xc420240000, 0x1860180, 0xc420432000, 0xc4208e8020, 0x17)
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/data_source_virtual_network.go:80 +0x53b
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).ReadDataApply(0xc4201b5200, 0xc42083e020, 0x1860180, 0xc420432000, 0xc4204d7220, 0xc4206b2301, 0x80000000018)
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:259 +0xb0
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).ReadDataApply(0xc420140a80, 0xc420172140, 0xc42083e020, 0x7feed9077458, 0x0, 0x74756f53222c2253)
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:398 +0x9a
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).ReadDataApply(0xc42013bfc0, 0xc4205a0010, 0xc4205a0040, 0x0, 0x0)
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:565 +0x4e
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: reflect.Value.call(0xc4201b58c0, 0xc420398368, 0x13, 0x18a1385, 0x4, 0xc4207d2f20, 0x3, 0x3, 0x302d35302d353130, 0x34313032222c2231, ...)
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: 	/usr/local/go/src/reflect/value.go:434 +0x905
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: reflect.Value.Call(0xc4201b58c0, 0xc420398368, 0x13, 0xc42080ff20, 0x3, 0x3, 0x73655270756f7247, 0x766f4d656372756f, 0x73736f7243202c65)
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: 	/usr/local/go/src/reflect/value.go:302 +0xa4
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: net/rpc.(*service).call(0xc4201707c0, 0xc420070050, 0xc42016c120, 0xc42017a400, 0xc42052c2c0, 0x15fcb60, 0xc4205a0010, 0x16, 0x15fcba0, 0xc4205a0040, ...)
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: 	/usr/local/go/src/net/rpc/server.go:381 +0x142
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: created by net/rpc.(*Server).ServeCodec
2018-02-06T09:11:31.207+0200 [DEBUG] plugin.terraform-provider-azurerm_v1.1.0_x4: 	/usr/local/go/src/net/rpc/server.go:475 +0x36b
2018/02/06 09:11:31 [ERROR] root: eval: *terraform.EvalReadDataApply, err: data.azurerm_virtual_network.elastic: unexpected EOF
2018/02/06 09:11:31 [ERROR] root: eval: *terraform.EvalSequence, err: data.azurerm_virtual_network.elastic: unexpected EOF
2018/02/06 09:11:31 [ERROR] root: eval: *terraform.EvalRefresh, err: azurerm_resource_group.elasticsearch: unexpected EOF
2018/02/06 09:11:31 [ERROR] root: eval: *terraform.EvalSequence, err: azurerm_resource_group.elasticsearch: unexpected EOF
2018/02/06 09:11:31 [TRACE] [walkRefresh] Exiting eval tree: azurerm_resource_group.elasticsearch
2018-02-06T09:11:31.208+0200 [DEBUG] plugin: plugin process exited: path=/elasticsearch-cloud-deploy/terraform-azure/.terraform/plugins/linux_amd64/terraform-provider-azurerm_v1.1.0_x4
2018/02/06 09:11:31 [TRACE] [walkRefresh] Exiting eval tree: data.azurerm_virtual_network.elastic
2018/02/06 09:11:31 [ERROR] root: eval: *terraform.EvalReadDataApply, err: data.azurerm_image.kibana: unexpected EOF
2018/02/06 09:11:31 [ERROR] root: eval: *terraform.EvalSequence, err: data.azurerm_image.kibana: unexpected EOF
2018/02/06 09:11:31 [TRACE] [walkRefresh] Exiting eval tree: data.azurerm_image.kibana
...
...
2018/02/06 09:11:31 [DEBUG] plugin: waiting for all plugin processes to complete...
2018-02-06T09:11:31.209+0200 [WARN ] plugin: error closing client during Kill: err="connection is shut down"
```

Version info:

```
Terraform v0.11.1
+ provider.azurerm v1.1.0
+ provider.random v1.1.0
+ provider.template v1.0.0
```

 I have the full crash report and can send it on a private channel if necessary

--- a/azurerm/data_source_virtual_network.go
+++ b/azurerm/data_source_virtual_network.go
@@ -77,9 +77,11 @@ func dataSourceArmVnetRead(d *schema.ResourceData, meta interface{}) error {
 			return err
 		}
 
-		dnsServers := flattenVnetAddressPrefixes(props.DhcpOptions.DNSServers)
-		if err := d.Set("dns_servers", dnsServers); err != nil {
-			return err
+		if options := props.DhcpOptions; options != nil {
+			dnsServers := flattenVnetAddressPrefixes(options.DNSServers)
+			if err := d.Set("dns_servers", dnsServers); err != nil {
+				return err
+			}
 		}
 
 		subnets := flattenVnetSubnetsNames(props.Subnets)

